src/evo/specialtx.cpp

```diff
@@ -37,6 +40,8 @@ bool CheckSpecialTx(const CTransaction& tx, const CBlockIndex* pindexPrev, CVali
             return CheckCbTx(tx, pindexPrev, state);
         case TRANSACTION_QUORUM_COMMITMENT:
             return llmq::CheckLLMQCommitment(tx, pindexPrev, state);
+        case TRANSACTION_MNHF_SIGNAL:
+            return CheckMNHFTx(tx, pindexPrev, state);
```

Comment:
`CheckSpecialTx` should NOT allow these new mnhf txes until the next hf (`DEPLOYMENT_GOV_FEE`) or it's a potential chain split otherwise - 0.17 nodes have no idea about these txes and 0.18 nodes would consider them valid.

Reply:
Good point!

Reply:
Thank you for pointing this, code has been updated

Reply:
Hmm.. Dropping these lines is not a solution either cause 0.18 nodes won't accept these txes now (which means they will fork off once such a tx is mined by some future version which in its turn makes this PR pointless). Should be smth like `return VersionBitsTipState(Params().GetConsensus(), Consensus::DEPLOYMENT_GOV_FEE) == ThresholdState::ACTIVE && CheckMNHFTx(tx, pindexPrev, state);` instead.

Reply:
okay, got it. I was wondering how it can be tested in v18.0. Thank you for letting me know.